Source/ACE/Factories/GeneratorFactory.cs

```diff
@@ -65,34 +66,34 @@ public static List<WorldObject> CreateWorldObjectsFromGenerator(AceObject genera
             }
 
             // Check the probability of this generator spawning something at all
-            if (random.Next(1, 100) >= generator.GeneratorProbability)
+            if (random.Next(1, 100) >= generator.GetIntProperty(PropertyInt.GeneratorProbability))
                 return null;
 
             // Generate objects from this generator #MaxGeneratedObjects times
-            for (int i = 0; i < generator.MaxGeneratedObjects; i++)
+            for (int i = 0; i < generator.GetIntProperty(PropertyInt.MaxGeneratedObjects); i++)
             {
-                switch (generator.GeneratorType)
+                switch (generator.GetIntProperty(PropertyInt.GeneratorType))
                 {
                     // Use the position of the generator as a static position
                     case (int)GeneratorType.Absolute:
-                        pos = generator.Location.InFrontOf(2.0);
+                        pos = generator.GetPosition(PositionType.Location).InFrontOf(2.0);
```

Comment:
the position of the generator object is moved backwards 2.0 prior to this code line, so it's just setting the object's position to the correct spot.